scripts/new-example.ts

```diff
@@ -0,0 +1,271 @@
+}
+
+(async function newExample() {
+    await promptForMissingArguments();
+    createFiles();
+    await registerWithDocumentItemsService();
+    if (args.requiredPackages!.length) {
+        installAdditionalPackages();
+    }
+    runBuild();
+
+    console.log(chalk.green('All done! Check your pending changes for TODOs for completing your example.'));
+})();
+
+function createFiles() {
+    const newExampleDir = join(examplesDir, args.name);
+    if (!existsSync(newExampleDir)) {
+        mkdirSync(newExampleDir);
+    }
+
+    const exampleComponentFileName = join(newExampleDir, `${args.name}-example.component.ts`);
+    console.info(chalk.gray(`creating ${exampleComponentFileName}...`));
+    const componentFileContents = `import {Component} from '@angular/core';
+
+@Component({
+selector: 'hc-${args.name}-example',
+templateUrl: '${args.name}-example.component.html',
+// TODO: delete the SCSS file if you don't need it in the example
+styleUrls: ['${args.name}-example.component.scss']
+})
+export class ${pascalCase(args.name)}ExampleComponent {
+// TODO: implement your example here
+}
+`;
+    writeFileSync(exampleComponentFileName, prettier.format(componentFileContents, {...prettierConfig, parser: 'typescript'}));
+
+    const exampleHtmlFileName = join(newExampleDir, `${args.name}-example.component.html`);
+    console.info(chalk.gray(`creating ${exampleHtmlFileName}...`));
+    writeFileSync(exampleHtmlFileName, `\n`);
+
+    const exampleScssFileName = join(newExampleDir, `${args.name}-example.component.scss`);
+    console.info(chalk.gray(`creating ${exampleScssFileName}...`));
+    writeFileSync(exampleScssFileName, `\n`);
+
+    if (args.type === 'module') {
+        const exampleModuleFileName = join(newExampleDir, `${args.name}-example.module.ts`);
+        const componentName = `${pascalCase(args.name)}ExampleComponent`;
+        console.info(chalk.gray(`creating ${exampleModuleFileName}...`));
+        const moduleFileContents =
+            `import {NgModule} from '@angular/core';
+import {CommonModule} from '@angular/common';
+import {CashmereModule} from '../cashmere.module';
+import {${componentName}} from './${args.name}-example.component';
+
+@NgModule({
+imports: [CommonModule, CashmereModule],
+declarations: [${componentName}],
+exports: [${componentName}],
+entryComponents: [${componentName}]
+})
+export class ${pascalCase(args.name)}ExampleModule {
+}
+`.trim() + '\n';
+        writeFileSync(exampleModuleFileName, prettier.format(moduleFileContents, {...prettierConfig, parser: 'typescript'}));
+    }
+}
+
+async function registerWithDocumentItemsService() {
+    console.info(chalk.gray(`registering example with document items service...`));
+    const docItemsDirectory = join(__dirname, '../src/app/core');
+    let docItemsFile: string;
+    if (args.category === 'cashmere') {
+        docItemsFile = join(docItemsDirectory, './cashmere-components-document-items.json');
```

Comment:
I found that manipulating JSON files was easier than manipulating TypeScript source files.  As such, all example metadata has been moved to JSON files.